src/search/search_handler.cpp

```diff
@@ -210,6 +225,13 @@ inline std::string searchTypeToString(uint8 type)
 
 void search_handler::read_func(uint16_t length)
 {
+    // if we already have a query in-flight...
+    if (!searchPackets.empty())
+    {
+        do_write();
+        return;
```

Comment:
Not a c guy, but shouldn't the method return a code to indicate it was unable to perform a read and instead needed to defer to write the awaiting packets to the socket (or otherwise could not perform what was asked of it)?

Would it be more ideal to make that the responsibility of the caller?

* Try to read
* Get told you can't due to x
* Resolve x however you need to
* Try to read again

Reply:
Error handling is done in the
`search_handler::do_write()` or `search_handler::do_write()` lambda. If it hits `search_handler::read_func()` that means there is NO error, which is either 

1) The clients initial request
2) The client's reply to our sending of search data